### ISSUE TITLE
Prevent text overlapping scaled scatter points

### ISSUE THREAD

**Issue by Trotts** [Phlya/adjustText#181]
Hi, 

I recently came across this repo as a potential solution to my current issue. I am building a scatterplot of values which contains some x, y values along with some scalar which dictates the size of each point. I am also required to label each point with some number, whereby extra information is provided in the legend based on this. 

I am trying to use this package so that when I add the textual labels to each point, each value is clearly visible (no overlap with other labels or the points). However, when using `adjust_text`, this does not seem to be operating as expected. I've tried following #76 as a guide, adding in `add_objects = ax.get_children()[0]` but this does not seem to be helping. 

For reference, here is the code I am currently using to generate a plot:

```
# Perform linear regression on the log-transformed data
log_class_count = np.log(tpr_per_class['Class Count']).values.reshape(-1, 1)
model = LinearRegression()
model.fit(log_class_count, tpr_per_class['True Positive Rate'])

# Predict values for the regression line
predicted_tpr = model.predict(log_class_count)

# Plot the scatter plot
fig, ax = plt.subplots(figsize=(10, 10))

# Plot the regression line
sorted_indices = np.argsort(tpr_per_class['Class Count'])
ax.plot(tpr_per_class['Class Count'][sorted_indices], predicted_tpr[sorted_indices], color='red')

labels = []

# Build points and labels
for i, label in enumerate(tpr_per_class.index):    
    # Scale the points by the average bbox area
    ax.scatter(tpr_per_class['Class Count'][i], tpr_per_class['True Positive Rate'][i],
               s=tpr_per_class['Average Bbox Area'][i]*1000,
               c = 'C0', edgecolors= "black", linewidth = 0.5)
    labels.append(f'{label}\nn={tpr_per_class["Class Count"][i]}, A={tpr_per_class["Average Bbox Area"][i]:.2f}')
    
# Use the adjustText package to label the points without overlap
plt.xscale('log')
txts = [ax.text(tpr_per_class['Class Count'][i], tpr_per_class['True Positive Rate'][i], i) for i in range(len(tpr_per_class))]
adjust_text(txts, add_objects = ax.get_children()[0])

#Make legend display the index rather than a point
#Source: https://stackoverflow.com/a/47382270
class TextHandler(HandlerBase):
    def create_artists(self, legend, orig_handle,xdescent, ydescent,
                        width, height, fontsize,trans):
        h = copy.copy(orig_handle)
        h.set_position((width/2.,height/2.))
        h.set_transform(trans)
        h.set_ha("center");h.set_va("center")
        fp = orig_handle.get_font_properties().copy()
        fp.set_size(fontsize)
        h.set_font_properties(fp)
        return [h]
handlermap = {type(txts[0]):TextHandler()}

# Make a red horizontal line using ax.text for the legend
line = ax.text(10, 10, '|', color='red', rotation = -50, fontsize = 0)
# Put at pos 14 so it appears at the end of the legend
txts.insert(14, line)
labels.insert(14, 'Log Line of Best Fit')

ax.legend(txts, labels, handler_map=handlermap,
                loc = 'center right', bbox_to_anchor = (1.68, 0.5), ncol = 2)
plt.xlabel('Class Count')
plt.ylabel('True Positive Rate')
plt.title('True Positive Rate (Test) vs Log Class Count (Train).\nPoint size is proportional to average bounding box area relative to patch size.')
plt.show()
```

When not using `adjust_text` I get the following output:

![image](https://github.com/Phlya/adjustText/assets/9531671/b91cd10d-6e01-4c31-b7ce-15b6dcac4281)

When using `adjust_text`, the labels' positions are changed only slightly:

![image](https://github.com/Phlya/adjustText/assets/9531671/e1abb379-e385-4322-94ed-a2ed20dfecd7)

Any advice is greatly appreciated, as ideally I would like to be able to automate the creation of these graphs based on incoming data.

**Comment by Phlya** [comment]
You don't need to use the `add_objects = ax.get_children()[0]` trick if you are not using seaborn, just use the output of ax.scatter.
Also, the argument has been renamed to 'objects' instead of 'add_objects'... For me with some random test data it (mostly) works quite well.
```python
x, y, s = np.random.random((3,30))
fig, ax = plt.subplots()
pathcollection = ax.scatter(x, y, s=s**2*1000)
texts = [ax.text(x[i], y[i], f'{i}') for i in range(len(x))]
adjust_text(texts, objects=pathcollection, arrowprops=dict(arrowstyle='-', color='red'));
```
![image](https://github.com/Phlya/adjustText/assets/2895034/23588bdb-712f-47be-9087-f906f4401f8e)

Occasionally when the patches are very close or overlap it doesn't do perfectly, but mostly it works.

**Comment by Trotts** [comment]
Hi, 

Thank you for the clarification! I've managed to get the code working as expected using the above plus some forcing:

```
adjust_text(txts, objects = path_collection, force_static= (0.1, 2.5),
            arrowprops=dict(arrowstyle="->", color='black', lw=0.5),
            min_arrow_len = 10)
```
